src/main.cpp

```diff
@@ -2512,24 +2517,28 @@ bool CBlock::ConnectBlock(CTxDB& txdb, CBlockIndex* pindex, bool fJustCheck, boo
             return error("ConnectBlock[] : %s unable to get coin age for coinstake", vtx[1].GetHash().ToString().substr(0,10).c_str());
 
         //9-3-2015
-        double dMaxResearchAgeReward = CoinToDouble(NN::ResearchAgeComputer::MaxReward(nTime));
+        int64_t nMaxResearchAgeReward = NN::ResearchAgeComputer::MaxReward(nTime);
```

Comment:
Me too. I rewrote the interface for `IAccrualComputer` in upcoming changes to remove the time parameter so that other implementations don't imply that the result is based on time. Only the `ResearchAgeComputer` retains the original calculation internally&mdash;I left the time-tiered calculation in for historical interest even though it has no impact on v9+ blocks as you said.